server/app.js

```diff
@@ -47,6 +62,20 @@ app.use('/courses', coursesRouter);
 app.use('/scraping', scriptRouter);
 app.use('/auth', authRouter);
 
+// Serving the frontend on wildcard. NEEDS TO REMAIN ABOVE 404 error
+if (process.env.NODE_ENV === 'production') {
+  app.get('/*', (req, res) => {
+    res.sendFile(path.join(__dirname, '..', 'frontend', 'build', 'index.html'));
+  });
+} else if (doesBuildDirExist) {
+  // checks to be able to see server endpoints in development
+  app.get('/*', (req, res) => {
+    // probably will never serve the generator files gain
+    // res.sendFile(path.join(__dirname, '..', 'frontend', 'public', 'index.html'));
+    res.sendFile(path.join(__dirname, '..', 'frontend', 'build', 'index.html'));
+  });
```

Comment:
same as above, I also don't see the difference between these two logic branches

Reply:
And here I uncomment this in the else:
```
res.sendFile(path.join(__dirname, '..', 'frontend', 'public', 'index.html'));
```

Reply:
There appears to be a difference in this codeblock, where there is no fallback else-statement after doesBuildDirExist; not sure if that was meant to be a thing, so I wanted to bring it up. Otherwise tested, and works; LGTM!

Reply:
i still think you should use or, i'll approve after you make this change, unless there's an actual real difference between these lines that i'm not seeing.